v2/internal/resolver/resource_hierarchy.go

```diff
@@ -261,6 +274,18 @@ func (h ResourceHierarchy) fullyQualifiedARMIDImpl(subscriptionID string, origin
 	}
 }
 
+func (h ResourceHierarchy) matchOwnerSubscription(subscriptionID string, ownerRID *arm.ResourceID) error {
```

Comment:
minor:
1. This method doesn't need to be on `ResourceHierarchy`, as it doesn't seem to use `h` at all. It could just be a package scoped function.
2. The name `matchOwnerSubscription` seems too specific to me. In reality this is just checking if the provided subscription matches the passed in ResourceID, so possibly we could rename to something like `checkARMIDHasSubscription` or `checkARMIDMatchesSubscription` or similar?

Reply:
Yep, I like it. Considering to move the method into `genruntime/arm_id.go` for re-usability.